interfaces/builtin/kernel_module.go

```diff
@@ -0,0 +1,48 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+var kernelModuleConnectedPlugAppArmor = `
+# Description: Allow insertion of modules.
+
+  capability sys_module,
+`
+
+var kernelModuleConnectedPlugSecComp = `
+# Description: Allow insertion of modules.
+
+finit_module
+delete_module
+`
+
+// NewKernelModuleInterface returns a new "kernel-module" interface.
+func NewKernelModuleInterface() interfaces.Interface {
+	return &commonInterface{
+		name: "kernel-module",
+		connectedPlugAppArmor: kernelModuleConnectedPlugAppArmor,
+		connectedPlugSecComp:  kernelModuleConnectedPlugSecComp,
+		reservedForOS:         true,
+		autoConnect:           true,
```

Comment:
No, you have to connect the plug and slot manually. Once hooks are available you can implement a hook in your snap so that you know when the plug got connected and then load the module.